### PR TITLE
Adds conditional templating for flowcontrol apiVersion

### PR DESCRIPTION
**Pull request by rewanthtammana**
Signed-off-by: Rewanth Tammana <22347290+rewanthtammana@users.noreply.github.com>
Towards https://github.com/giantswarm/giantswarm/issues/25865
### This PR:

- Adds conditional templating for `flowcontrol` apiVersion value

### Checklist

- [x] Update changelog in CHANGELOG.md.
- [x] Make sure `values.yaml` and `values.schema.json` are valid.

### CODE REVIEW

File: helm/trivy-operator/.kube-linter.yaml [hunk 1]
```diff
@@ -16,3 +16,8 @@ checks:
   - "minimum-three-replicas"  # Upstream doesn't specify, and only 1 is currently needed.
   - "no-rolling-update-strategy"  # Upstream doesn't include strategy.
   - "non-isolated-pod"
+  - "no-node-affinity" # Node affinity not required. We can have the pod scheduled on any node.
+  - "dnsconfig-options" # No explicity dns configuration required.
+  # Path exclusion should be working according to the documentation but it's not working for some reason
+  # ignorePaths:
+  #   - ./helm/trivy-operator/charts/*
```
Comment (stone-z) [inline helm/trivy-operator/.kube-linter.yaml:23]: Can we take this out, then?

File: helm/trivy-operator/templates/networkpolicy.yaml [hunk 2]
```diff
@@ -5,6 +5,8 @@ kind: NetworkPolicy
 metadata:
   name: {{ include "trivy-operator.networkPolicy" . }}-networkpolicy
   labels: {{- include "trivy-operator-helpers.labels" . | nindent 4 }}
+  annotations:
+    ignore-check.kube-linter.io/dangling-networkpolicy: "The podselctor value, app.kubernetes.io/managed-by:trivy-operator is inherited from parent repository."
```
Comment (stone-z) [inline helm/trivy-operator/templates/networkpolicy.yaml:9]:
```suggestion
    ignore-check.kube-linter.io/dangling-networkpolicy: "The podselector value, app.kubernetes.io/managed-by:trivy-operator is inherited from parent repository."
```

File: helm/trivy-operator/templates/networkpolicy.yaml [hunk 3]
```diff
@@ -35,6 +37,8 @@ kind: NetworkPolicy
 metadata:
   name: {{ include "trivy-operator.networkPolicy" . }}-scanner-job-networkpolicy
   labels: {{- include "trivy-operator-helpers.labels" . | nindent 4 }}
+  annotations:
+    ignore-check.kube-linter.io/dangling-networkpolicy: "The podselctor value, app.kubernetes.io/managed-by:trivy-operator is inherited from parent repository."
```
Comment (stone-z) [inline helm/trivy-operator/templates/networkpolicy.yaml:41]:
```suggestion
    ignore-check.kube-linter.io/dangling-networkpolicy: "The podselector value, app.kubernetes.io/managed-by:trivy-operator is inherited from parent repository."
```

**Review by stone-z — COMMENTED** [review]
Isn't it redundant to have the annotations and the `.kube-linter` file in this case?

**Comment by rewanthtammana** [comment]
If the `ignorePaths` work, the best way would be to go with annotations as it gives granular control over exceptions & we get to ignore upstream configuration violations.

File: helm/trivy-operator/.kube-linter.yaml (hunk 1 above)
Comment (rewanthtammana) [reply helm/trivy-operator/.kube-linter.yaml:23]: Left it out for future reference. It's not working as intended for now.

**Comment by rewanthtammana** [comment]
Despite excluding all files in `.kube-linter.yaml`, the paths aren't being excluded. Likely an upstream bug.

```yaml
checks:
  addAllBuiltIn: true
  exclude:
  - "no-read-only-root-fs"
  - "required-annotation-email"
  - "required-label-owner"
  - "no-liveness-probe"
  - "no-readiness-probe"
  - "unset-memory-requirements"
  - "access-to-create-pods"
  - "unset-cpu-requirements"
  - "run-as-non-root"
  - "access-to-secrets"
  - "env-var-secret" # False positive. Upstream uses an environment variable named 'OPERATOR_EXPOSED_SECRET_SCANNER_ENABLED'.
  - "use-namespace"  # Upstream doesn't include namespace.
  - "minimum-three-replicas"  # Upstream doesn't specify, and only 1 is currently needed.
  - "no-rolling-update-strategy"  # Upstream doesn't include strategy.
  - "non-isolated-pod"
  ignorePaths:
  - ./*
  - ./**
```

image

File: CHANGELOG.md [hunk 4]
```diff
@@ -11,6 +11,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Changed
 
+- Add conditional templating for `flowcontrol` apiVersion value.
+
+### Changed
```
Comment (stone-z) [inline CHANGELOG.md:16]: Changelog is a bit wonky. This should be under unreleased

File: helm/trivy-operator/templates/crd-install/crd-job.yaml [hunk 5]
```diff
@@ -8,6 +8,8 @@ metadata:
     # create hook dependencies in the right order
     "helm.sh/hook-weight": "-1"
     {{- include "trivy-operator.CRDInstallAnnotations" . | nindent 4 }}
+    ignore-check.kube-linter.io/no-node-affinity: "Node affinity not required. We can have the pod scheduled on any node."
+    ignore-check.kube-linter.io/dnsconfig-options: "No explicity dns configuration required."
```
Comment (stone-z) [inline helm/trivy-operator/templates/crd-install/crd-job.yaml:12]:
```suggestion
    ignore-check.kube-linter.io/dnsconfig-options: "No explicit dns configuration required."
```

File: helm/trivy-operator/.kube-linter.yaml [hunk 6]
```diff
@@ -16,3 +16,8 @@ checks:
   - "minimum-three-replicas"  # Upstream doesn't specify, and only 1 is currently needed.
   - "no-rolling-update-strategy"  # Upstream doesn't include strategy.
   - "non-isolated-pod"
+  - "no-node-affinity" # Node affinity not required. We can have the pod scheduled on any node.
+  - "dnsconfig-options" # No explicity dns configuration required.
```
Comment (stone-z) [inline helm/trivy-operator/.kube-linter.yaml:20]:
```suggestion
  - "dnsconfig-options" # No explicit dns configuration required.
```

**Review by stone-z — APPROVED** [review]

**Pull request merged**